app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -527,70 +529,76 @@ class HomeFragment : Fragment(), CoroutineScope, AccountObserver {
+                                Tab(
+                                    it.id,
+                                    it.url,
+                                    it.url.urlToTrimmed(context!!),
+                                    it.title,
+                                    selected,
+                                    it.thumbnail
+                            )
+                        }
+                )
             )
-        )
-
-        deleteSessionJob = {
-            sessionManager.findSessionById(sessionId)
-                ?.let { session ->
-                    sessionManager.remove(session)
-                }
-        }
 
-        allowUndo(
-            view!!, getString(R.string.snackbar_tab_deleted),
-            getString(R.string.snackbar_deleted_undo), {
-                deleteSessionJob = null
-                emitSessionChanges()
+            deleteSessionJob = {
+                sessionManager.findSessionById(sessionId)
+                    ?.let { session ->
+                        sessionManager.remove(session)
+                    }
             }
-        ) {
-            sessionManager.findSessionById(sessionId)
-                ?.let { session ->
-                    sessionManager.remove(session)
+
+            allowUndo(
+                view!!, getString(R.string.snackbar_tab_deleted),
+                getString(R.string.snackbar_deleted_undo), {
+                    deleteSessionJob = null
+                    emitSessionChanges()
                 }
+            ) {
+                sessionManager.findSessionById(sessionId)
+                    ?.let { session ->
+                        sessionManager.remove(session)
+                    }
+            }
         }
     }
 
     private fun emitSessionChanges() {
-        val sessionManager = requireComponents.core.sessionManager
-
-        getManagedEmitter<SessionControlChange>().onNext(
-            SessionControlChange.TabsChange(
-                sessionManager.sessions
-                    .filter { (activity as HomeActivity).browsingModeManager.isPrivate == it.private }
-                    .map {
-                        val selected = it == sessionManager.selectedSession
-                        Tab(
-                            it.id,
-                            it.url,
-                            it.url.urlToTrimmedHost(),
-                            it.title,
-                            selected,
-                            it.thumbnail
-                        )
-                    }
-            )
-        )
+        context?.let { context ->
+            launch(Dispatchers.Main) {
+                val sessionManager = requireComponents.core.sessionManager
+
+                getManagedEmitter<SessionControlChange>().onNext(
+                    SessionControlChange.TabsChange(
+                        sessionManager.sessions
+                            .filter { (activity as HomeActivity).browsingModeManager.isPrivate == it.private }
```

Comment:
Nit: put 

```kotlin
(activity as HomeActivity).browsingModeManager.isPrivate
```

into a local variable.

Reply:
What's the benefit in doing this 🤔

Reply:
It's a nit, so I'm not insanely worried about it. However:

 * The expression can be calculated once, but it's currently being calculated for every session in the `sessionManager`. It's possible that the compiler knows that both the `browsingModeManager` and `isPrivate` properties are both side effect free, so can optimize it for you, but I don't know.

 * From a style point of view: having a cast and then two dots in a small closure makes the line long — and in narrow windows like the diff viewer — slightly harder to read :(